### ISSUE TITLE
Reject "connect" request when user closes popup

### ISSUE THREAD

**Issue by marsrobertson** [MetaMask/metamask-extension#6489]
Mod edit: when a user closes an eip1102 "Connect" request, we can provide an error/callback message to the dapp (as we now do with the transaction/confirm window)

---

Demo: https://youtu.be/oy6zN9NM_r4

```
try {
  accounts = await ethereum.enable()
} catch (error) {
  console.log(error)
}
```

Closing the window should give some callback message.

Otherwise some sloppy UI may wait for callback that never comes.

**Comment by bdresser** [comment]
Handled for regular tx in https://github.com/MetaMask/metamask-extension/issues/3124

As stated in https://github.com/MetaMask/metamask-extension/issues/3124#issuecomment-437032657 this can apply to the 1102 Connect request as well

Previously filed as https://github.com/MetaMask/metamask-extension/issues/5700

@bitpshr @danjm any concerns?

**Comment by marsrobertson** [comment]
You have much better searching skills than me, I was expecting this is likely to cause some outcry, feel free to mark as duplicate.

https://github.com/MetaMask/metamask-extension/issues/5700#issuecomment-437034583:

> This means that when a user closes an approval request rather than responding to it, it's the same as if they minimized it: the `Promise` will continue to hang, waiting for either an explicit approval or rejection.

**This is bad.**

I was thinking about some timeout long-polling to as a workaround to this bug.

Nope - let the website if the window was closed - the intention of the user was clear - closing window means no.

Minimizing window... That could trigger some event too. 
```
if(metamask minimised) then 
     1. hide spinner
     2. display notice "metamask minimised"
```

**Comment by jtrein** [comment]
@marsrobertson Did you find an _interim_ solution to this? I was also thinking about long-polling, but reluctant.

**Comment by jtrein** [comment]
I did notice that when the popup is triggered open, the event `web3Instance.currentProvider.publicConfigStore.on("update", callback)` will emit every ~20000ms. If it stops emitting, then we can assume (most likely unsafely) that the popup window has been closed by the user.

**Comment by tmashuang** [comment]
Reopening issue, the linked PR doesn't seem to implement Reject on window close.

**Comment by Gudahtt** [comment]
This was fixed by #7401
